portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -21,26 +21,52 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.Arrays;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  Gson gson = new Gson();  
+private static Gson gson = new Gson();  
+private static final ArrayList<String> comments = new ArrayList<String>();  
   
  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> jsonData = new ArrayList<String>();  
-    jsonData.add("nice!");
-    jsonData.add("cool!");
-    jsonData.add("awesome!");
-    String jsonOutput = convertToJsonUsingGson(jsonData);
     response.setContentType("application/json;");
-    response.getWriter().println(jsonOutput);
+    response.getWriter().println(convertToJsonUsingGson(comments));
   }
 
   private String convertToJsonUsingGson(ArrayList<String> jsonData) {
     return gson.toJson(jsonData);
   }
+  
+  // Get the comments from the user and display them
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String input = getParameter(request, "text-input", "");
+    String[] words = input.split("\\s*,\\s*");
+
+    int i = 0;
+    for ( String word : words) {
```

Comment:
This type of for loop means that the code within the braces executes for every element in `words`. You would access the value of the current word with the `word` variable that you declared in the for loop.

In other words, you wouldn't need the `i` variable and you can just use `comments.add(word + "\n")`

Reply:
Understood, even though its called "for each" for some reason I thought I still had to iterate with i